tsconfig.json

```
@@ -0,0 +1,33 @@
+{
+    "compileOnSave": true,
+    "compilerOptions": {
+        "noFallthroughCasesInSwitch": true,
+        "allowUnreachableCode": false,
+        "allowUnusedLabels": false,
+        "strict": true,
+        "strictNullChecks": false,
+        "noImplicitAny" : false,
+        "preserveSymlinks": true,
+        "allowJs": true,
+        "resolveJsonModule": true,
+        "sourceMap": true,
+        "declaration": true,
+        "inlineSources": true,
+        "typeRoots": ["./node_modules/@types","./lib/types" ],
+        "rootDir": "./",
```

I don't thing we need any files within the root directory no?
```suggestion
        "rootDir": "./lib",
```